core/src/banking_stage/worker.rs

```diff
@@ -0,0 +1,631 @@
+use {
+    super::{
+        consumer::{Consumer, ExecuteAndCommitTransactionsOutput, ProcessTransactionBatchOutput},
+        forwarder::Forwarder,
+        scheduler_messages::{ConsumeWork, FinishedConsumeWork, FinishedForwardWork, ForwardWork},
+        ForwardOption,
+    },
+    crossbeam_channel::{select, Receiver, RecvError, SendError, Sender},
+    solana_poh::leader_bank_notifier::LeaderBankNotifier,
+    solana_runtime::bank::Bank,
+    std::{sync::Arc, time::Duration},
+    thiserror::Error,
+};
+
+#[derive(Debug, Error)]
+pub enum WorkerError {
+    #[error("Failed to receive work from scheduler: {0}")]
+    Recv(#[from] RecvError),
+    #[error("Failed to send finalized consume work to scheduler: {0}")]
+    ConsumedSend(SendError<FinishedConsumeWork>),
+    #[error("Failed to send finalized forward work to scheduler: {0}")]
+    ForwardedSend(SendError<FinishedForwardWork>),
+}
+
+impl From<SendError<FinishedConsumeWork>> for WorkerError {
+    fn from(err: SendError<FinishedConsumeWork>) -> Self {
+        Self::ConsumedSend(err)
+    }
+}
+
+impl From<SendError<FinishedForwardWork>> for WorkerError {
+    fn from(err: SendError<FinishedForwardWork>) -> Self {
+        Self::ForwardedSend(err)
+    }
+}
```

Comment:
nits: these `From` impls can be removed as well.

Reply:
36facc9770f71a46e07a1b8c0ba0a5c157464cc8